bloom-instance/cloudfront/main.tf

```diff
@@ -0,0 +1,144 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.64"
+    }
+  }
+}
+locals {
+  qualified_name = "${var.name_prefix}-${var.name}"
+
+  distribution       = var.distribution
+  cloudfront_enabled = local.distribution.enabled
+
+  # Only certain combinations are valid for allowed methods
+  allowed_method_map = {
+    "get"          = ["HEAD", "GET"]
+    "with-options" = ["HEAD", "GET", "OPTIONS"]
+    "all"          = ["HEAD", "DELETE", "POST", "GET", "OPTIONS", "PUT", "PATCH"]
+  }
+
+  cached_method_map = {
+    "get"          = ["HEAD", "GET"]
+    "with-options" = ["HEAD", "GET", "OPTIONS"]
+  }
+
+  # Origin
+  origin_id           = "${local.qualified_name}-default"
+  origin_alb_name     = local.distribution.origin.alb
+  origin_alb          = var.alb_map[local.origin_alb_name]
+  origin_alb_dns_name = local.origin_alb.dns_name
+
+  # Certificates
+  cert_arn = var.distribution.certificate.arn
+  # cert_arn  = var.cert_map[local.cert_name]
+
+
+  # The default cache has the same data shape, but is treated differently
+  default_cache = local.distribution.cache.default
+  # The other cache rules need to be separated out
+  other_caches = { for path, cache in local.distribution.cache : path => cache if path != "default" }
+
+  # Cache policies that need to be created
+  cache_policies_to_create = {
+    for name, cache in local.distribution.cache : name => cache.policy if cache.policy != null
+  }
+
+  # The policy IDs that were provided directly
+  provided_cache_policy_ids = {
+    for name, cache in local.distribution.cache : name => cache.policy_id if cache.policy_id != null
+  }
+
+  # The policy IDs for policies we created
+  generated_cache_policy_ids = {
+    for name, policy in module.policies : name => policy.id
+  }
+
+  # All policy IDs, provided and generated
+  policy_ids = merge(local.provided_cache_policy_ids, local.generated_cache_policy_ids)
+}
+
+module "policies" {
+  source   = "./policy"
+  for_each = local.cache_policies_to_create
+
+  name_prefix = local.qualified_name
+  policy      = each.value
+}
+
+
+resource "aws_cloudfront_distribution" "main" {
+  origin {
+    origin_id   = local.origin_id
+    domain_name = local.origin_alb.dns_name
+
+    custom_origin_config {
+      http_port              = 80
+      https_port             = 443
+      origin_protocol_policy = "http-only"
+      origin_ssl_protocols   = ["SSLv3", "TLSv1", "TLSv1.1", "TLSv1.2"]
+    }
+  }
+
+  # Derived from known values
+  comment = local.qualified_name
+
+  # Configurable
+  enabled     = local.distribution.enabled
+  aliases     = local.distribution.domains
+  price_class = local.distribution.price_class
+
+  # TODO: add logging config
```

Comment:
Is this todo still applicable?

Reply:
Missed this but lets keep it for now. Another pull request is coming w/ a universal S3 module in a few days and I'd rather use that module.